### PR TITLE
Feature/game page

### LINKED ISSUE #115: As a game enthusiast, I want to be able to read more about the games, so that I can learn more about them
- [x] Add an endpoint to fetch data for a single game
- [x] Create a game page
- [x] Implement the design

### PR DESCRIPTION
**Pull request by Supergamer1337**
## Key Features
- Introduces a game detail page!
- Added enum for status codes, hopefully leading to less misuse.
- Made markAsPlayed optimistically updated.
- Closes #115 

## Known Issues
- Mark game as played can cover the description if a. the description is too long, or b. the screen not wide enough. I could probably fix this, but I'm tired, want to sleep, and don't see it as that important right now.
- Rate game doesn't look great, but that's because we lack the Reviews feature, and therefore I didn't think putting it somewhere else would be worth it.
- Clicking on a game card doesn't open the detail screen, as I think it best to add it as part of the home page redesign.

### Other notes
- I'm sorry for my obsessive compulsion to refactor everything I come across 😢 

### Screenshot of the page
image

### CODE REVIEW

**Comment by Oscariremma** [comment]
Maybe we want som kind of confirmation page before actually deleting a game? 
Having a button that just immediately deletes feels a bit scarry....

File: backend/src/routers/gameRouter.ts [hunk 1]
```diff
@@ -174,51 +174,38 @@ gameRouter.post(
...
-		try {
-			if (!req.user) {
-				return res
-					.status(401)
-					.json({ message: 'Must be logged in to add game' });
-			}
+		if (!req.user)
+			return res
+				.status(StatusCode.Unauthorized)
+				.json({ message: 'Must be logged in to add game' });
```
Comment (Oscariremma) [inline backend/src/routers/gameRouter.ts:180]: This should be replaced with the `isAuthenticated` middleware

File: backend/src/routers/gameRouter.ts [hunk 2]
```diff
@@ -244,23 +231,20 @@ gameRouter.post(
...
 gameRouter.delete('/:id', async (req, res) => {
-	try {
-		if (!req.user) {
-			res.status(401).json({ message: 'Must be logged in to remove game' });
-			return;
-		}
+	if (!req.user)
+		return res
+			.status(StatusCode.Unauthorized)
+			.json({ message: 'Must be logged in to remove game' });
```
Comment (Oscariremma) [inline backend/src/routers/gameRouter.ts:237]: Same here, this should be replaced with the `isAuthenticated` middleware

File: backend/src/services/gameService.ts [hunk 3]
```diff
@@ -115,6 +100,20 @@ export const getGameById = async (gameId: string) => {
 	});
 };
 
+export const getExtendedGameById = async (id: string) => {
+	// I don't really like the game of this function, but it works
```
Comment (Oscariremma) [inline backend/src/services/gameService.ts:104]: ?
```suggestion
	// I don't really like the name of this function, but it works
```

File: frontend/src/components/GameDetails/GamePropertiesCard/GamePropertiesCard.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+							: playerMin.toString()
+					}
+				/>
+				<Property
+					title="Availability"
+					value={borrowed ? 'Borrowed' : 'Not borrowed'}
+				/>
+			</div>
+			<BorrowGame gameId={gameId} />
+		</div>
```
Comment (Oscariremma) [inline frontend/src/components/GameDetails/GamePropertiesCard/GamePropertiesCard.tsx:52]: We should add the games average rating somewhere. Maybe here?

**Review by Oscariremma — CHANGES_REQUESTED** [review]
Overall it looks really good! Just some small things that need changing.
One thing I noticed is that we don't display the games average rating anywhere, we should probably do that.

File: frontend/src/components/GameDetails/GamePropertiesCard/GamePropertiesCard.tsx (hunk 4 above)
Comment (Supergamer1337) [reply frontend/src/components/GameDetails/GamePropertiesCard/GamePropertiesCard.tsx:52]: Not actually part of the design that I worked from, so it's kind of @Krysset that needs to answer for this.

**Comment by Oscariremma** [comment]
> > Maybe we want som kind of confirmation page before actually deleting a game? Having a button that just immediately deletes feels a bit scarry....
> 
> The delete button is actually not part of the design atm, but was something I just added since we don't have an edit page. I think we should add a modal at a later date though too.
> 
> If you feel very strongly about it though, I could add an alert or something with the built-in modal

Ok, then it is fine. 
Using the built in modal works great for now

**Comment by Oscariremma** [comment]
> > Overall it looks really good! Just some small things that need changing. One thing I noticed is that we don't display the games average rating anywhere, we should probably do that.
> 
> That isn't part of the the design actually, I wondered about that myself.

Oh, that's interesting... We probably forgot about it when designing.
Ether we change the design to add it in this PR or we open another issue about adding it later.

**Comment by Supergamer1337** [comment]
> > > Overall it looks really good! Just some small things that need changing. One thing I noticed is that we don't display the games average rating anywhere, we should probably do that.
> > 
> > 
> > That isn't part of the the design actually, I wondered about that myself.
> 
> Oh, that's interesting... We probably forgot about it when designing. Ether we change the design to add it in this PR or we open another issue about adding it later.

I personally think that we should just wait with it to another PR as we are kind of pressed on time, and I would like to have it all done by the scrum meeting tomorrow (or even better, by the stakeholder meeting)

**Comment by Oscariremma** [comment]
> > > > Overall it looks really good! Just some small things that need changing. One thing I noticed is that we don't display the games average rating anywhere, we should probably do that.
> > > 
> > > 
> > > That isn't part of the the design actually, I wondered about that myself.
> > 
> > 
> > Oh, that's interesting... We probably forgot about it when designing. Ether we change the design to add it in this PR or we open another issue about adding it later.
> 
> I personally think that we should just wait with it to another PR as we are kind of pressed on time, and I would like to have it all done by the scrum meeting tomorrow (or even better, by the stakeholder meeting)

Ok, then we need to open an issue about it.

**Review by Oscariremma — APPROVED** [review]

**Review by Krysset — APPROVED** [review]
Looks good! As was mentioned earlier a few things missing due to the design not being entirely complete. I agree that we fix that in an another PR

**Pull request merged**